src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/HearingDocumentController.java

```diff
@@ -75,6 +77,20 @@ public ResponseEntity<AboutToStartOrSubmitCallbackResponse> generateHearingDocum
         }
 
         List<String> warnings = validateHearingService.validateHearingWarnings(caseDetails);
+
+        // NOTE TO SELF, TEST BOTH PAPER AND DIGITAL JOURNEYS
+        if ((!notificationService.shouldEmailRespondentSolicitor(caseDetails.getData())
+            || !notificationService.shouldEmailContestedAppSolicitor(caseDetails.getData()))
+            && caseDataService.isContestedApplication(caseDetails)) {
```

Comment:
Personally i think reordering conditions in this if statement would benefit from `short circuit` boolean evaluation. i.e  moving `caseDataService.isContestedApplication(caseDetails))` to the start of `if` would not check `||` if this returns false.

Reply:
hey Faisal, Michael and myself have checked this over and we think based on the logical groupings that this statement isn't a candidate for short circuit boolean evaluations. It is ugly however we can't see another way to slice it.